File: src/main/java/com/exzell/exzlvideoplayer/adapter/tracker/MediaKeyProvider.java
package com.exzell.exzlvideoplayer.adapter.tracker;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.selection.ItemKeyProvider;
import androidx.recyclerview.widget.RecyclerView;

import com.exzell.exzlvideoplayer.adapter.VideoFileAdapter;

public class MediaKeyProvider extends ItemKeyProvider {

    private RecyclerView mRecyclerView;

    public MediaKeyProvider(RecyclerView rv) {
        super(ItemKeyProvider.SCOPE_MAPPED);
        mRecyclerView = rv;
    }

    @Nullable
    @Override
    public Object getKey(int position) {
        long itemId = mRecyclerView.getAdapter().getItemId(position);
        return itemId == -1 ? RecyclerView.NO_ID : itemId;
    }

    @Override
    public int getPosition(@NonNull Object key) {

        VideoFileAdapter.ViewHolder vh = (VideoFileAdapter.ViewHolder) mRecyclerView.findViewHolderForItemId((Long) key);
        return vh == null ? RecyclerView.NO_POSITION : vh.getAdapterPosition();
    }
}

File: app/src/main/java/com/exzell/exzlvideoplayer/adapter/tracker/MediaKeyProvider.kt
package com.exzell.exzlvideoplayer.adapter.tracker


import androidx.recyclerview.selection.ItemKeyProvider
import androidx.recyclerview.widget.RecyclerView

import com.exzell.exzlvideoplayer.adapter.VideoFileAdapter

import androidx.recyclerview.selection.ItemKeyProvider.*

class MediaKeyProvider(private val mRecyclerView: RecyclerView) : ItemKeyProvider<Long>(SCOPE_MAPPED) {

    override fun getKey(position: Int): Long? {
        val itemId = mRecyclerView.adapter!!.getItemId(position)
        return if (itemId == RecyclerView.NO_ID) RecyclerView.NO_ID else itemId
    }

    override fun getPosition(key: Long): Int {

        val vh = mRecyclerView.findViewHolderForItemId(key) as? VideoFileAdapter.ViewHolder
        return vh?.absoluteAdapterPosition ?: RecyclerView.NO_POSITION
    }
}
